lxd/api_project.go

```diff
@@ -1078,6 +1078,104 @@ func projectNodeConfigDelete(d *Daemon, s *state.State, name string) error {
 	return nil
 }
 
+// doProjectForceDelete handles force deletion of project entities.
+func doProjectForceDelete(ctx context.Context, s *state.State, projectName string) error {
+	// Get the project entities.
+	var usedBy []string
+	err := s.DB.Cluster.Transaction(ctx, func(ctx context.Context, tx *db.ClusterTx) error {
+		project, err := dbCluster.GetProject(ctx, tx.Tx(), projectName)
+		if err != nil {
+			return fmt.Errorf("Failed loading project %q: %w", projectName, err)
+		}
+
+		usedBy, err = projectUsedBy(ctx, tx, project)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Parse entity URLs.
+	defaultProfile := api.NewURL().Path(version.APIVersion, "profiles", api.ProjectDefaultName).Project(projectName).String()
+	toDelete := make([]entity.Reference, 0, len(usedBy))
+	for _, u := range usedBy {
+		if u == defaultProfile {
```

Comment:
this should have a comment i think - especially as in theory the user deleting the project might not have `auth.EntitlementCanDelete` on it, and yet is allowed to delete the project and its default profile.

cc @markylaing